framework/src/timesteppers/IterationAdaptiveDT.C

```diff
@@ -201,7 +212,21 @@ IterationAdaptiveDT::preExecute()
 Real
 IterationAdaptiveDT::computeInitialDT()
 {
-  return _input_dt;
+  Real dt;
+  if (_tfunc_last_step)
+  {
+    dt = _time_ipol.sample(_time_old);
+    if (_verbose)
+      _console << "Setting initial dt to value specified by dt function: " << std::setw(9) << dt
+               << std::endl;
+  }
+  else
+  {
+    dt = _input_dt;
+    if (_verbose)
+      _console << "Setting initial dt to input value: " << std::setw(9) << dt << std::endl;
+  }
+  return dt;
```

Comment:
This change now respects the t/dt function list pair that is passed in input files, rather than just taking the initial dt.